client.go

```diff
@@ -103,53 +104,68 @@ func httpRespWithRetries(c context.Context, req *Request) (*http.Response, error
 	}
 	var httpResp *http.Response
 	var err error
-	for i := 1; i <= req.maxAttempts; i++ {
+	for i := 1; ; i++ {
 		req.debugf("request attempt #%d", i)
 		httpResp, err = req.client.client.Do(reqc)
-		if err != nil {
+		if err != nil && err != io.EOF {
```

Comment:
I'm wondering if this could be put into a separate error checker.

```
if err != nil && req.isErrARealErr(c, err) {...

// #needsbettername
func (req *Request) isErrARealErr(c, err) bool {
  if err == io.EOF && req.retryOnEOFError {
    return false
  }
  return true
}
```